js/tools.js

```diff
@@ -0,0 +1,334 @@
+let selectButton = document.getElementById("select");
+let lineButton = document.getElementById("line");
+
+/**
+ * This class represents the Canvas
+ */
+class Canvas {
+    constructor(canvasId, canvasContainerId) {
+        // svg canvas
+        this.canvas = document.getElementById(canvasId);  
+        // container for the canvas to support accurate mouse click coordinates
+        this.canvasContainer = document.getElementById(canvasContainerId);
+        // list of shapes drawn on the canvas; shapes must be of class Shape
+        this.shapes = [];  
+        this.currentShape = null;
```

Comment:
not too supportive for multiple shapes but i'm ok w/ that for now

Reply:
the shapes array is being used for selection purposes. Without it, it becomes very difficult to track down which shape has been selected, and also to which shape styles need to be addded